app/bot/openai.go

```diff
@@ -0,0 +1,94 @@
+package bot
+
+import (
+	"context"
+	"fmt"
+	"github.com/sashabaranov/go-openai"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type OpenAIClient interface {
+	CreateChatCompletion(context.Context, openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error)
+}
+
+// OpenAI bot, returns responses from ChatGPT via OpenAI API
+type OpenAI struct {
+	authToken string
+	client    OpenAIClient
+}
+
+// NewOpenAI makes a bot for ChatGPT
+func NewOpenAI(authToken string, httpClient *http.Client) *OpenAI {
+	log.Printf("[INFO] OpenAI bot with github.com/sashabaranov/go-openai")
+	config := openai.DefaultConfig(authToken)
+	config.HTTPClient = httpClient
+
+	client := openai.NewClientWithConfig(config)
+	return &OpenAI{authToken, client}
+}
+
+// Help returns help message
+func (o *OpenAI) Help() string {
+	return genHelpMsg(o.ReactOn(), "Спросите что-нибудь у ChatGPT")
+}
+
+func (o *OpenAI) chatGPTRequest(request string) (response string, err error) {
+	resp, err := o.client.CreateChatCompletion(
+		context.Background(),
+		openai.ChatCompletionRequest{
+			Model: openai.GPT3Dot5Turbo,
+			Messages: []openai.ChatCompletionMessage{
+				{
+					Role:    openai.ChatMessageRoleUser,
+					Content: request,
+				},
+			},
+		},
+	)
+
+	if err != nil {
+		log.Printf("[WARN] failed to make request to ChatGPT '%s', error=%v", request, err)
+		return "", err
+	}
+
+	if resp.Choices == nil || len(resp.Choices) == 0 {
```

Comment:
would be nice to have a comment here about what Choices filed evan mean, and why we pick choice 0. The one reading the code can be a little bit confused by this logic